docs/clients/service-providers.md

```diff
@@ -121,3 +121,49 @@ mechanism for instance.
 In order to generate an unsigned transaction (example with
 [coin transfer](https://cosmos.network/rpc/#/ICS20/post_bank_accounts__address__transfers)),
 you need to use the field `generate_only` in the body of `base_req`.
+
+## Cosmos signing
+
+
+Cosmos transaction signing is a fairly simple process. 
+
+Every Cosmos transaction has a canonical json representation. The gaiacli and stargate rest interfaces provides canonical json representations of transactions and their broadcast function will provide compact amino(a protobuf like format) encoding translations.
+
+
+Things to know when signing messages.
+
+The format is 
+```json
+{ 
+"account_number": XXX,
+"chain_id":XXX,
+"fee":XXX,
+"sequence":XXX 
+"memo":XXX,
+"msgs":XXX
+}
+```
+
+The signer must supply chain_id, account number and sequence number.
+
+Fee, msg and memo will be supplied by the transaction composer interface.
+
+`account_number` and `sequence` can be queried directly from the blockchain or cached locally. Getting these numbers wrong is a common cause of invalid signature errors. You can load the mempool of a full node or validator with a sequence of uncommitted transactions with incrementing sequence numbers and it will mostly do the correct thing.  
```

Comment:
```suggestion
`"account_number"` and `"sequence"` can be queried directly from the blockchain or cached locally. Getting these numbers wrong is a common cause of invalid signature errors. You can load the mempool of a full node or validator with a sequence of uncommitted transactions with incrementing sequence numbers and it will mostly do the correct thing.  
```